spec/models/miq_schedule_spec.rb

```diff
@@ -511,6 +511,33 @@
       end
     end
 
+    context "valid action_automation_request" do
+      let(:admin) { FactoryGirl.create(:user_miq_request_approver) }
+      before(:each) do
+        @automate_sched = MiqSchedule.create(:name          => "test_method", :towhat => "AutomationRequest",
+                                             :userid        => admin.userid, :enabled => true,
+                                             :run_at        => {:interval   => {:value => "1", :unit => "daily"},
+                                                                :start_time => 2.hours.from_now.utc.to_i},
+                                             :sched_action  => {:method => "automation_request"},
+                                             :filter        => {:uri_parts  => {:namespace => 'ss',
+                                                                                :instance  => 'vv',
+                                                                                :message   => 'mm'},
+                                                                :parameters => {"param" => "8"}})
+      end
+
+      it "should create a request from a scheduled task" do
+        allow(AutomationRequest).to receive(:create_from_ws).and_return(AutomationRequest.new)
+        expect(AutomationRequest).to receive(:create_from_scheduled_task).once
+        @automate_sched.run_automation_request
+      end
+
+      it "should create 1 automation request" do
+        @automate_sched.action_automation_request(AutomationRequest, '')
+        request = AutomationRequest.where(:description => "Automation Task", :userid => admin.userid)
+        expect(request.length).to eq(1)
```

Comment:
I think it would be slightly more efficient to merge this expect with the line above and use `#count`.
```
expect(AutomationRequest.where(:description => "Automation Task", :userid => admin.userid).count).to eq(1)
```